core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowExecutionsResource.scala

```diff
@@ -104,4 +104,47 @@ class WorkflowExecutionsResource {
     )
       throw new WebApplicationException(Response.Status.UNAUTHORIZED);
   }
+
+  /** Delete a single execution */
+  @GET
+  @Path("/{wid}-{eid}")
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  def retrieveExecutionsOfWorkflow(
+      @PathParam("wid") wid: UInteger,
+      @PathParam("eid") eid: UInteger,
+      @Auth sessionUser: SessionUser
+  ): List[WorkflowExecutionEntry] = {
+    /* delete the execution in sql */
+    context
+      .delete(WORKFLOW_EXECUTIONS)
+      .where(WORKFLOW_EXECUTIONS.WID.eq(wid))
+      .and(WORKFLOW_EXECUTIONS.EID.eq(eid))
+      .execute();
+
+    /* re-display the workflow execution after deletion */
```

Comment:
since retrieving list from database again is expensive, especially that it has a join, I suggest you only perform the deletion, and if successful, you just delete the entry in the frontend. Follow similar logic of bookmark.